prov/util/src/util_av.c

```diff
@@ -553,7 +553,11 @@ int ofi_av_init_lightweight(struct util_domain *domain, const struct fi_av_attr
 	av->context = context;
 	av->domain = domain;
 
-	ret = ofi_genlock_init(&av->ep_list_lock, OFI_LOCK_MUTEX);
+	ret = ofi_genlock_init(&av->ep_list_lock,
+			       (av->domain->threading == FI_THREAD_DOMAIN ||
+			       av->domain->threading == FI_THREAD_COMPLETION) &&
+			       av->domain->control_progress == FI_PROGRESS_CONTROL_UNIFIED ?
+					OFI_LOCK_NOOP : OFI_LOCK_MUTEX);
```

Comment:
Please separate out this set of checks (repeated 3 times) into a function somewhere.  E.g. ofi_lockless_progress()?  I'm unsure of the name.

Otherwise, I think this works, is still simple to use, and has minimal impact on providers.

Reply:
done